src/tokens.md

```diff
@@ -758,19 +758,34 @@ r[lex.token.life.syntax]
 > &nbsp;&nbsp; &nbsp;&nbsp; `'` [IDENTIFIER_OR_KEYWORD][identifier]
 >   _(not immediately followed by `'`)_\
 > &nbsp;&nbsp; | `'_`
->   _(not immediately followed by `'`)_
+>   _(not immediately followed by `'`)_\
+> &nbsp;&nbsp; | RAW_LIFETIME
 >
 > LIFETIME_OR_LABEL :\
 > &nbsp;&nbsp; &nbsp;&nbsp; `'` [NON_KEYWORD_IDENTIFIER][identifier]
 >   _(not immediately followed by `'`)_\
 > &nbsp;&nbsp; | `'_`
+>   _(not immediately followed by `'`)_\
+> &nbsp;&nbsp; | RAW_LIFETIME
+>
+> RAW_LIFETIME :\
+> &nbsp;&nbsp; `'r#` [IDENTIFIER_OR_KEYWORD][identifier]
```

Comment:
I believe this should also include `_`, right? That is, `'r#_` is valid, correct?

I also just want to double-check, it is intentional that `'r#_` is allowed as a loop label? (I only ask because `'_` is explicitly not allowed).

```suggestion
> &nbsp;&nbsp; &nbsp;&nbsp; `'r#` [IDENTIFIER_OR_KEYWORD][identifier]
>   _(not immediately followed by `'`)_\
> &nbsp;&nbsp; | `'r#_`
```

Reply:
Actually this is likely something *else* we want to deny here.  I feel like `'r#_` is invalid, or at least not something we need to support initially.

Reply:
https://github.com/rust-lang/rust/pull/132363

Hopefully resolved by https://github.com/rust-lang/reference/pull/1603/commits/912a6d5c0c04ea77a90fcdbb5ca8fe7a5f12f6b9